### PR TITLE
ci: improve maintenance of test containers

### LINKED ISSUE #2422: Improve arch test container
Install an [aur helper](https://wiki.archlinux.org/title/AUR_helpers) to make it easier to install aur packages (such us tgt)

Install the following packages to enable more testing (just like the Fedora container)
- bash-completion                                                                                                                                                                                          
- biosdevname                                                                                                                                                                                              
- bluez                                                                                                                                                                                                    
- busybox                                                                                                                                                                                                  
- bzip2                                                                                                                                                                                                    
- cifs-utils                                                                                                                                                                                               
- cryptsetup                                                                                                                                                                                               
- e2fsprogs                                                                                                                                                                                                
- f2fs-tools                                                                                                                                                                                               
- fuse3                                                                                                                                                                                                    
- iproute                                                                                                                                                                                                  
- iputils                                                                                                                                                                                                  
- kbd                                                                                                                                                                                                      
- lzop                                                                                                                                                                                                     
- memstrack                                                                                                                                                                                                
- ndctl                                                                                                                                                                                                    
- nvme-cli                                                                                                                                                                                                 
- rng-tools                                                                                                                                                                                                
- sbsigntools                                                                                                                                                                                              
- tar                                                                                                                                                                                                      
- tgt                                                                                                                                                                                                      
- tpm2-tools

### PR DESCRIPTION
**Pull request by Henrik66**
Install AUR helper (paru) into Arch test container and improve maintenance of test containers.

Pulls in https://github.com/dracutdevs/dracut/pull/2427 as one of the commits in this PR.

## Checklist
- [X] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #2422 #2433

### CODE REVIEW

**Review by LaszloGombos — COMMENTED** [review]
Thanks @Henrik66 . 

Can you please also add `pcsclite` package ? 

Can you see if you can find a way not to install `mkinitcpio` (the default `initramfs` provider on arch) ? There is a possibly solution used in the `Debian` container to avoid having `initramfs-tools` installed.

File: test/container/Dockerfile-Arch [hunk 1]
```diff
@@ -2,16 +2,64 @@ FROM docker.io/archlinux
...
+RUN su build -c 'cd && git clone https://aur.archlinux.org/paru-bin.git && cd paru-bin && makepkg -s --noconfirm' && pacman -U --noconfirm ~build/paru-bin/*.pkg.tar.*
 
-RUN useradd -m build
-RUN su build -c 'cd && git clone https://aur.archlinux.org/perl-config-general.git && cd perl-config-general && makepkg -s --noconfirm'
-RUN pacman -U --noconfirm ~build/perl-config-general/*.pkg.tar.*
-RUN su build -c 'cd && git clone https://aur.archlinux.org/tgt.git && cd tgt && makepkg -s --noconfirm'
-RUN pacman -U --noconfirm ~build/tgt/*.pkg.tar.*
-RUN rm -fr ~build
+# Install needed packages for the dracut CI container
+RUN su build -c 'paru --noconfirm -Syu \
```
[the remark this answers is not in the document]
Comment (Henrik66) [reply test/container/Dockerfile-Arch]: Ok. Done.

File: test/container/Dockerfile-Arch [hunk 2]
```diff
@@ -2,16 +2,64 @@ FROM docker.io/archlinux
...
+    linux \
+    lvm2 \
+    lzop \
+    mdadm \
+    memstrack \
+    multipath-tools \
+    nbd \
+    ndctl \
+    networkmanager \
+    nfsidmap \
```
[the remark this answers is not in the document]
Comment (Henrik66) [reply test/container/Dockerfile-Arch]: Ok. Done.

File: test/container/Dockerfile-Arch [hunk 3]
```diff
@@ -2,16 +2,64 @@ FROM docker.io/archlinux
...
+    cpio \
+    cryptsetup \
+    dash \
+    dhclient \
+    dhcp \
+    dmraid \
+    e2fsprogs \
+    f2fs-tools \
+    fuse3 \
+    glibc \
```
[the remark this answers is not in the document]
Comment (Henrik66) [reply test/container/Dockerfile-Arch]: Ok. Done.

**Comment by Henrik66** [comment]
> Can you see if you can find a way not to install `mkinitcpio` (the default `initramfs` provider on arch) ? There is a possibly solution used in the `Debian` container to avoid having `initramfs-tools` installed.

I found out that [assume-installed](https://archlinux.org/pacman/pacman.8.html#:~:text=%2D%2D-,assume%2Dinstalled,-%3Cpackage%3Dversion%3E) option can do this.

**Review by LaszloGombos — APPROVED** [review]
Thanks

**Comment by Henrik66** [comment]
Start working on https://github.com/dracutdevs/dracut/issues/2433 .

Pulling in https://github.com/dracutdevs/dracut/pull/2427 as this PR would help with https://github.com/dracutdevs/dracut/issues/2433 .

Marking in draft for now as I will likely need help with this PR.

**Review by LaszloGombos — APPROVED** [review]
LGTM. Hopefully this helps with maintenance - and at least now we have a list of common packages installed in the containers.

**Comment by bdrung** [comment]
The test code is harder to read and harder to copy-paste after this change.

The list would be easier to maintain if the entries would be grouped by topic. Example question: Why is btrfs-progs not installed on openSUSE? There are many packages that just have different names. Having those in one place would help, but I have no idea how to codify that in a readable way.

```
opensuse-tumbleweed: ShellCheck, others: shellcheck
fedora|opensuse-tumbleweed: kernel, debian|ubuntu: linux-image-generic
```

**Comment by bdrung** [comment]
> Some documentation also references package names - https://github.com/dracutdevs/dracut/blob/master/docs/README.generic . We should either remove these or replace them with some syntax that considers several distributions.

Is this list referring to projects or package names?

Also "Generic initramfs'es are huge (usually over 10 megs in size uncompressed)" is a little outdated. The generic initramfs is more around 100 MB and uncompressed in hundreds of megabytes.

**Comment by pvalena** [comment]
LGTM, although I've not verified the package lists.

**Pull request closed without merging**